app/uk/gov/hmrc/agentsubscription/model/OverseasAgentApplication.scala

```diff
@@ -18,29 +18,93 @@ package uk.gov.hmrc.agentsubscription.model
 import play.api.libs.functional.syntax._
 import play.api.libs.json.{ Json, Reads, _ }
 
-case class TradingDetails(tradingName: String, tradingAddress: OverseasBusinessAddress)
-
-case class OverseasContactDetails(businessTelephone: String, businessEmail: String)
+sealed trait OverseasAddress {
+  def addressLine1: String
+  def addressLine2: String
+  def addressLine3: Option[String]
+  def addressLine4: Option[String]
+  def countryCode: String
+}
 
-case class OverseasAgencyDetails(
-  agencyName: String,
-  agencyEmail: String,
-  agencyAddress: OverseasAgencyAddress)
+final case class UkAddressForOverseas(
+  addressLine1: String,
+  addressLine2: String,
+  addressLine3: Option[String],
+  addressLine4: Option[String],
+  postalCode: String,
+  countryCode: String) extends OverseasAddress
```

Comment:
since this extends `OverseasAddress` , i think you can simply have only one field  here `countryCode` you dont need to declare every other field?